src/test/regress/expected/alter_database_from_nonmain_db.out

```diff
@@ -0,0 +1,401 @@
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+\c test_alter_db_from_nonmain_db - - :worker_2_port
+set citus.log_remote_commands = true;
+set citus.grep_remote_commands = "%ALTER DATABASE%";
+alter database "altered_database!'2"  rename to altered_database_renamed;
+NOTICE:  issuing SELECT citus_internal.execute_command_on_remote_nodes_as_user('ALTER DATABASE "altered_database!''2" RENAME TO altered_database_renamed', 'postgres')
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+alter database altered_database_renamed  rename to "altered_database!'2";
+NOTICE:  issuing SELECT citus_internal.execute_command_on_remote_nodes_as_user('ALTER DATABASE altered_database_renamed RENAME TO "altered_database!''2"', 'postgres')
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+alter database "altered_database!'2" with
+    ALLOW_CONNECTIONS true
+    CONNECTION LIMIT 0
+    IS_TEMPLATE false;
+NOTICE:  issuing SELECT citus_internal.execute_command_on_remote_nodes_as_user('ALTER DATABASE "altered_database!''2" WITH ALLOW_CONNECTIONS true CONNECTION LIMIT 0 IS_TEMPLATE false;', 'postgres')
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+\c regression - - :worker_2_port
+set citus.log_remote_commands = true;
+set citus.grep_remote_commands = "%ALTER DATABASE%";
+alter database "altered_database!'2" with
+    ALLOW_CONNECTIONS false
+    CONNECTION LIMIT 1
+    IS_TEMPLATE true;
+NOTICE:  issuing ALTER DATABASE "altered_database!'2" WITH ALLOW_CONNECTIONS false CONNECTION LIMIT 1 IS_TEMPLATE true;
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing ALTER DATABASE "altered_database!'2" WITH ALLOW_CONNECTIONS false CONNECTION LIMIT 1 IS_TEMPLATE true;
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+alter database "altered_database!'2" with
+    ALLOW_CONNECTIONS true
+    CONNECTION LIMIT 0
+    IS_TEMPLATE false;
+NOTICE:  issuing ALTER DATABASE "altered_database!'2" WITH ALLOW_CONNECTIONS true CONNECTION LIMIT 0 IS_TEMPLATE false;
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing ALTER DATABASE "altered_database!'2" WITH ALLOW_CONNECTIONS true CONNECTION LIMIT 0 IS_TEMPLATE false;
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+\c regression
+create role test_owner_non_main_db;
+\c test_alter_db_from_nonmain_db
+set citus.log_remote_commands = true;
+set citus.grep_remote_commands = "%ALTER DATABASE%";
+set citus.enable_create_database_propagation=on;
+alter database "altered_database!'2" owner to test_owner_non_main_db;
+NOTICE:  issuing SELECT citus_internal.execute_command_on_remote_nodes_as_user('ALTER DATABASE "altered_database!''2" OWNER TO test_owner_non_main_db;', 'postgres')
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+\c test_alter_db_from_nonmain_db - - :worker_1_port
```

Comment:
Can you also add checks to see the result, for example, after the alter database ... owner to ... , connect to another db / worker node and check the result, e.g. select db.datname, pg_catalog.pg_get_userbyid(db.datdba) FROM pg_catalog.pg_database db where db.datname = ....;

Reply:
Alternatively, can use the functions that we used before to collect settings of a database from all nodes (via run_command_on_all_nodes etc.), but yes, we definitely somehow need to check database options on all nodes after issuing those commands from a non-main db.

Reply:
Added value checks for tests